api/clients/v2/payload_disperser.go

```diff
@@ -0,0 +1,222 @@
+	codec           codecs.BlobCodec
+	disperserClient DisperserClient
+	certVerifier    verification.ICertVerifier
+}
+
+// BuildPayloadDisperser builds a PayloadDisperser from config structs
+func BuildPayloadDisperser(
+	logger logging.Logger,
+	payloadDisperserConfig *PayloadDisperserConfig,
+	disperserClientConfig *DisperserClientConfig,
+	// signer to sign blob dispersal requests
+	signer core.BlobRequestSigner,
+	// prover is used to compute commitments to a new blob during the dispersal process
+	//
+	// IMPORTANT: it is permissible for the prover parameter to be nil, but operating with this configuration
+	// puts a trust assumption on the disperser. With a nil prover, the disperser is responsible for computing
+	// the commitments to a blob, and the PayloadDisperser doesn't have a mechanism to verify these commitments.
+	//
+	// TODO: In the future, an optimized method of commitment verification using fiat shamir transformation will
+	//  be implemented. This feature will allow a PayloadDisperser to offload commitment generation onto the
+	//  disperser, but the disperser's commitments will be verifiable without needing a full-fledged prover
+	prover encoding.Prover,
+	accountant *Accountant,
+	ethConfig geth.EthClientConfig,
+) (*PayloadDisperser, error) {
+
+	codec, err := codecs.CreateCodec(
+		payloadDisperserConfig.PayloadPolynomialForm,
+		payloadDisperserConfig.BlobEncodingVersion)
+	if err != nil {
+		return nil, fmt.Errorf("create codec: %w", err)
+	}
+
+	disperserClient, err := NewDisperserClient(disperserClientConfig, signer, prover, accountant)
+	if err != nil {
+		return nil, fmt.Errorf("new disperser client: %s", err)
+	}
+
+	ethClient, err := geth.NewClient(ethConfig, gethcommon.Address{}, 0, logger)
+	if err != nil {
+		return nil, fmt.Errorf("new eth client: %w", err)
+	}
+
+	certVerifier, err := verification.NewCertVerifier(*ethClient, payloadDisperserConfig.EigenDACertVerifierAddr)
+	if err != nil {
+		return nil, fmt.Errorf("new cert verifier: %w", err)
+	}
+
+	return &PayloadDisperser{
+		logger:          logger,
+		config:          payloadDisperserConfig,
+		codec:           codec,
+		disperserClient: disperserClient,
+		certVerifier:    certVerifier,
+	}, nil
+}
+
+// SendPayload executes the dispersal of a payload, with these steps:
+//
+//  1. Encode payload into a blob
+//  2. Disperse the blob
+//  3. Poll the disperser with GetBlobStatus until a terminal status is reached, or until the polling timeout is reached
+//  4. Construct an EigenDACert if dispersal is successful
+//  5. Verify the constructed cert with a call to an ethereum contract
+//  6. Return the valid cert
+func (pd *PayloadDisperser) SendPayload(
+	ctx context.Context,
+	// payload is the raw data to be stored on eigenDA
+	payload []byte,
+	// salt is added while constructing the blob header
+	// This salt should be utilized if a blob dispersal fails, in order to retry dispersing the same payload under a
+	// different blob key.
```

Comment:
only true for reserved bandwidth payments.

Reply:
Clarified [c08d16f8](https://github.com/Layr-Labs/eigenda/pull/1159/commits/c08d16f8e7b62275763a018dd9fc153284303f86)